src/generate/utils/State_AI.cpp

```diff
@@ -48,4 +52,123 @@ namespace spy::gameplay {
+            successChance /= 2;
+        }
+
+        // if char has tradecraft and not mole die, prob. test is repeated
+        if (character.hasProperty(PropertyEnum::TRADECRAFT)
+            && !character.hasGadget(GadgetEnum::MOLEDIE)) {
+            successChance = 1 - std::pow(1 - successChance, 2);
+        }
+
+        stateChance *= successChance;
+    }
+
+    std::pair<std::vector<State_AI>, bool> State_AI::handleHoneyTrap(const GadgetAction &op, const MatchConfig &config,
+                                                                     const libclient::LibClient &libClient) {
+        std::vector<State_AI> honeyTrapStates;
+        bool foundAlternatives = false;
+
+        auto sourceChar = this->getCharacters().findByUUID(op.getCharacterId());
+        auto targetChar = util::GameLogicUtils::findInCharacterSetByCoordinates(this->getCharacters(),
+                                                                                op.getTarget());
+
+        if (!this->wasHoneyTrapUsed && config.getHoneyTrapSuccessChance() != 0 &&
+            targetChar != this->getCharacters().end()) {
+            if (targetChar->hasProperty(spy::character::PropertyEnum::HONEY_TRAP)) {
+                auto otherOps = this->getHoneyTrapAlternatives(op, config);
+                if (!otherOps.empty()) {
+                    foundAlternatives = true;
+
+                    // copy and modify state
+                    State_AI s = *this;
+                    s.wasHoneyTrapUsed = true;
+                    s.modStateChance(*sourceChar,
+                                     config.getHoneyTrapSuccessChance() / static_cast<double>(otherOps.size()));
+
+                    // reset stuff from executeGadget base method
+                    s.operationsLeadingToState.pop_back();
+                    s.usedGadgets.pop_back();
+                    auto character = s.getCharacters().getByUUID(op.getCharacterId());
+                    character->setActionPoints(character->getActionPoints() + 1);
+                    s.isLeafState = false;
+
+                    // handle possible states due to honeytrap
+                    for (const auto &oOp: otherOps) {
+                        auto honeyTrapState = OperationExecutor::executeGadget(s, oOp, config, libClient);
+                        honeyTrapStates.insert(honeyTrapStates.end(), honeyTrapState.begin(), honeyTrapState.end());
+                    }
+                }
+            }
+        }
+
+        this->modStateChance(*sourceChar, (1 - config.getHoneyTrapSuccessChance()));
+        return std::pair<std::vector<State_AI>, bool>(honeyTrapStates,
+                                                      foundAlternatives && config.getHoneyTrapSuccessChance() == 1);
+    }
+
+    std::vector<GadgetAction>
+    State_AI::getHoneyTrapAlternatives(const GadgetAction &op, const MatchConfig &config) const {
+        GadgetAction a = op; // keep op const to know regular operation/target
```

Comment:
```suggestion
    State_AI::getHoneyTrapAlternatives(GadgetAction &a, const MatchConfig &config) const {
```

Reply:
```suggestion
    State_AI::getHoneyTrapAlternatives(GadgetAction a, const MatchConfig &config) const {
```
Wenn dann so, damit die GadgetAction kopiert wird. Ansonsten manipuliere ich meine ursprüngliche Operation, die ich aber noch weiter verarbeiten sollte.

Reply:
Habe das jetzt mal wie in meinem Vorschlag gemacht

Reply:
oh ups, ja das hatte ich natürlich genau so gemeint wie du das vorgeschlagen hast! :+1: